### DISCUSSION TITLE
Remote Access to Open Lite Speed

### DISCUSSION THREAD

**Discussion by ghost** [community/community#140929] [Programming Help]
### Body

I want my web server which is my pc using my mobile hotspot as a network to allow remote access and respond with an index.html file.

I have added two listeners http on port 8080 and https on port 8088. I have allowed these ports for incoming traffic using ufw. The listeners' ip address in any ipv4 address. On virtual host mapping I have put my public ipv4 address as the domain name as I currently do not have a domain. For the virtual host I have added my public ipv4 address again as the domain name. Now when I copy my public ipv4 address into the url alongisde the port it is saying this site cannot be reached. Now when i edit the configuration and all the parts where i was putting my public ipv4 address i now put my local ipv4 address it is now working but only for devices in my LAN. Can you help me guys to allow remote access to anyone not looking at the network he is in

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by R0b0t-Maker** [comment]
It sounds like you've configured your server to work within your local network (LAN) but are having issues with allowing access to it remotely (over the internet). Here are some steps and considerations to make your PC accessible over the internet using your mobile hotspot:

1.  Your public IP address might be behind your mobile carrier's NAT (Network Address Translation). This means that your mobile ISP might not allow direct access to your public IP address from the internet. You can check this by running `curl ifconfig.me` to get your public IP, and compare it with the IP shown on your hotspot configuration. If they are different, your ISP uses carrier-grade NAT (CGNAT), and you will need to work around that (explained later).
2. Some mobile hotspots allow for port forwarding, but most do not. You can check your hotspot settings to see if this feature is available. If it is, forward ports `8080` (HTTP) and `8088` (HTTPS) to the local IP address of your PC.
3. Since your public IP provided by the ISP might change frequently, you can use a dynamic DNS service (like No-IP, DynDNS, etc.) to map a domain name to your changing IP. If your public IP is static, this step can be skipped.
4.  Since you’ve already opened the necessary ports with `ufw`, ensure that your hotspot or ISP isn’t blocking those ports. Sometimes, mobile ISPs block common service ports.
